client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/trainedmodel/InferenceStats.java

```diff
@@ -0,0 +1,170 @@
+                          Instant timeStamp) {
+        this.missingAllFieldsCount = missingAllFieldsCount;
+        this.inferenceCount = inferenceCount;
+        this.failureCount = failureCount;
+        this.cacheMissCount = cacheMissCount;
+        this.timeStamp = timeStamp == null ?
+            Instant.ofEpochMilli(Instant.now().toEpochMilli()) :
+            Instant.ofEpochMilli(timeStamp.toEpochMilli());
+    }
+
+    /**
+     * How many times this model attempted to infer with all its fields missing
+     */
+    public long getMissingAllFieldsCount() {
+        return missingAllFieldsCount;
+    }
+
+    /**
+     * How many inference calls were made against this model
+     */
+    public long getInferenceCount() {
+        return inferenceCount;
+    }
+
+    /**
+     * How many inference failures occurred.
+     */
+    public long getFailureCount() {
+        return failureCount;
+    }
+
+    /**
+     * How many cache misses occurred when inferring this model
+     */
+    public long getCacheMissCount() {
+        return cacheMissCount;
+    }
+
+    /**
+     * The timestamp of these statistics.
+     */
+    public Instant getTimeStamp() {
+        return timeStamp;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(FAILURE_COUNT.getPreferredName(), failureCount);
+        builder.field(INFERENCE_COUNT.getPreferredName(), inferenceCount);
+        builder.field(CACHE_MISS_COUNT.getPreferredName(), cacheMissCount);
+        builder.field(MISSING_ALL_FIELDS_COUNT.getPreferredName(), missingAllFieldsCount);
+        builder.timeField(TIMESTAMP.getPreferredName(), TIMESTAMP.getPreferredName() + "_string", timeStamp.toEpochMilli());
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        InferenceStats that = (InferenceStats) o;
+        return missingAllFieldsCount == that.missingAllFieldsCount
+            && inferenceCount == that.inferenceCount
+            && failureCount == that.failureCount
+            && cacheMissCount == that.cacheMissCount
+            && Objects.equals(timeStamp, that.timeStamp);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(missingAllFieldsCount, inferenceCount, failureCount, cacheMissCount, timeStamp);
+    }
+
+    @Override
+    public String toString() {
+        return "InferenceStats{" +
+            "missingAllFieldsCount=" + missingAllFieldsCount +
+            ", inferenceCount=" + inferenceCount +
+            ", failureCount=" + failureCount +
+            ", cachMissCount=" + cacheMissCount +
+            ", timeStamp=" + timeStamp +
+            '}';
+    }
+
+    private static long unbox(@Nullable Long value) {
```

Comment:
It would be better to call this something like `unboxOrZero`, as it doesn't simply unbox.